templates/publisher/settings.html

```diff
@@ -0,0 +1,220 @@
+              <div class="js-whitelist">
+                <input type="radio" name="territories_custom_type" value="whitelist" id="territories_custom_whitelist"
+                  {% if whitelist_country_codes or not blacklist_country_codes %}
+                      checked="checked"
+                  {% endif %}
+                />
+                <label for="territories_custom_whitelist">Only these territories</label>
+                <input type="text" class="js-multiselect-input" name="whitelist_countries" value="
+                  {% for territory in whitelist_country_codes %}
+                    {{ territory }}{{ "," if not loop.last }}
+                  {% endfor %}
+                 " />
+                <div class="js-multiselect-holder{% if blacklist_country_codes %} is-disabled{% endif %}"></div>
+              </div>
+              <div class="js-blacklist">
+                <input type="radio" name="territories_custom_type" value="blacklist" id="territories_custom_blacklist"
+                  {% if blacklist_country_codes %}
+                       checked="checked"
+                  {% endif %}
+                />
+                <label for="territories_custom_blacklist">Exclude these territories</label>
+                <input type="text" class="js-multiselect-input" name="blacklist_countries" value="
+                  {% for territory in blacklist_country_codes %}
+                    {{ territory }}{{ "," if not loop.last }}
+                  {% endfor %}
+                 " />
+                <div class="js-multiselect-holder{% if whitelist_country_codes or not blacklist_country_codes %} is-disabled{% endif %}"></div>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div class="p-strip is-shallow">
+          <div class="row">
+            <hr class="u-no-margin--bottom" />
+          </div>
+        </div>
+      </section>
+    </form>
+  </div>
+{% endblock %}
+
+{% block scripts %}
+  <script src="{{ static_url('js/dist/publisher.js') }}"></script>
+  <script>
+    Raven.context(function () {
+      snapcraft.publisher.initMultiselect('.js-whitelist', {{ countries|tojson }});
+      snapcraft.publisher.initMultiselect('.js-blacklist', {{ countries|tojson }});
+
+      snapcraft.publisher.market.initForm({
+          form: "settings-form",
+          formNotification: "settings-form-status",
+        }, {
+          'private': {{ private|tojson }},
+          'whitelist_countries': {{ whitelist_country_codes|tojson }},
+          'blacklist_countries': {{ blacklist_country_codes|tojson }}
+        },
+        {% if error_list %}
+          {{ error_list|tojson}}
+        {% else %}
+          null
+        {% endif %}
+      );
+
+      document
+        .getElementById('settings-form')
+        .addEventListener(
+          'change',
+          snapcraft.publisher.settings.changeHandler
```

Comment:
Why this is separate from `initForm`? Does it handle different things that the form does? If this is not a market page why it uses `market.initForm`?

Reply:
I'm tempted to rename market.js to form.js... I wanted to reuse all the state logic you'd already written so I did, though it's true, this is not the market page.

Reply:
If it's shared makes sense to rename it - would be less confusing.

Reply:
Renamed.

The reason this is here is that is shows/hides the custom radio's based on what's selected. I didn't want to add yet more to the marketForm js.

I agree this should be refactored at a later date, now we have 2 separate forms using the same code.... Probably a good time to reactify it too.